plugins/utilities/src/deephaven/plugin/utilities/utils.py

```
@@ -0,0 +1,159 @@
+    """
+    Package the built JS files at the given JS directory and unpack them into the destination directory.
+
+    Args:
+        js_dir:
+            The directory containing the JS files
+        dest_dir:
+            The directory to unpack the JS files into
+    """
+    dist_dir = os.path.join(js_dir, "dist")
+    build_dir = os.path.join(js_dir, "build")
+    package_dir = os.path.join(build_dir, "package")
+
+    # copy the bundle to the directory
+    # the path may not exist (e.g. when running tests)
+    # so it is not strictly necessary to copy the bundle
+    if os.path.exists(dist_dir):
+        # ignore errors as the directory may not exist
+        shutil.rmtree(build_dir, ignore_errors=True)
+        shutil.rmtree(dest_dir, ignore_errors=True)
+
+        os.makedirs(build_dir, exist_ok=True)
+
+        # pack and unpack into the js directory
+        subprocess.run(
+            ["npm", "pack", "--pack-destination", "build"], cwd=js_dir, check=True
+        )
+        # it is assumed that there is only one tarball in the directory
+        files = os.listdir(build_dir)
+        for file in files:
+            subprocess.run(["tar", "-xzf", file], cwd=build_dir, check=True)
+            os.remove(os.path.join(build_dir, file))
+
+        # move the package directory to the expected package location
+        shutil.move(package_dir, dest_dir)
+
+
+def attempt_registration(callback: Callback, js_plugin: JsPlugin) -> None:
+    """
+    Attempt to register a JS plugin.
+    If failed, and enterprise is not detected, it will log as a warning and continue.
+
+    Args:
+        callback:
+            A callback to register the JS plugin
+        js_plugin:
+            The JS plugin to register
+    """
+    try:
+        callback.register(js_plugin)
+    except RuntimeError:
+        # warning should only be logged if enterprise is not detected
+        if not os.path.exists("/usr/illumon/latest/bin"):
+            logger.warning(
```

I think we should prefer to re-raise in this case instead of logging?

---

Ideally if this failed because the JS package was already registered it would be a warning as that could *possibly* be ok, but I'm sure there are plenty of other problematic errors that would be ignored. Do you know of any graceful way both could be handled?

---

I guess I'm not sure exactly what is causing the double registration (not familiar enough w/ DHE setup). That said, I can't think of a community deployment where we might be OK with just logging a double registration as opposed to throwing the error.

---

Yea I think Devin's right, we can throw in the Community side. Enterprise we probably should log a debug message, users won't see it by default.
@devinrsmith is there a better way to detect an Enterprise env? Pretty sure installation could be in a different directory.

---

The case I had in mind was if someone wanted a different version of the js then was stored in the wheel. But that is an edge development case so I am fine with the debug for enterprise and throw for community.